snippets/terraform/deployments/with-diagnostic-setting/main.tf

```diff
@@ -0,0 +1,59 @@
+terraform {
+  required_version = "~> 1.3"
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "~> 3.85"
+    }
+  }
+}
+
+provider "azurerm" {
+  features {}
+}
+
+module "prerequisites" {
+  source   = "../../prerequisites"
+  location = var.location
+  name     = var.name
+  tags     = var.tags
+}
+
+data "azurerm_storage_account" "example" {
+  name                = var.storage_account_name
+  resource_group_name = var.storage_account_resource_group
+}
+
+resource "azurerm_nginx_deployment" "example" {
+  name                     = var.name
+  resource_group_name      = module.prerequisites.resource_group_name
+  sku                      = var.sku
+  location                 = var.location
+  capacity                 = 50
+  diagnose_support_enabled = false
+
+  identity {
+    type = "SystemAssigned"
+  }
+
+  frontend_public {
+    ip_address = [module.prerequisites.public_ip_address_id]
+  }
+  network_interface {
+    subnet_id = module.prerequisites.subnet_id
+  }
+
+  tags = var.tags
+}
+
+resource "azurerm_monitor_diagnostic_setting" "example" {
+  name               = "${azurerm_nginx_deployment.example.name}-logs"
+  target_resource_id = azurerm_nginx_deployment.example.id
+  storage_account_id = data.azurerm_storage_account.example.id
+
+  enabled_log {
+    category = "NginxLogs"
+  }
+
+  depends_on = [ azurerm_role_assignment.nginx_identity_storage_contributor_role ]
```

Comment:
Does this need to depend on this role assignment?

Reply:
Nope. Another good catch! Just saw this in my test run (finally got through ARM templates and SDK...)